lib/gc/ExecutionEngine/GPURuntime/ocl/GpuOclRuntime.cpp

```diff
@@ -20,6 +20,40 @@
 #include "mlir/Dialect/LLVMIR/LLVMDialect.h"
 #include "mlir/Pass/PassManager.h"
 
+#ifdef GC_ENABLE_GPU_PROFILE
+
+void EnableProfiling();
+void DisableProfiling();
+
+class GPUKernelTracer {
+public:
+  static std::unique_ptr<GPUKernelTracer> getInstance() {
+    std::unique_ptr<GPUKernelTracer> instance =
+        std::make_unique<GPUKernelTracer>();
+    return instance;
+  }
+
+  GPUKernelTracer() {
+    gcLogD("Enable Profiling.");
+    EnableProfiling();
+  }
+
+  ~GPUKernelTracer() {
+    gcLogD("Profiling is finished.");
+    DisableProfiling();
+  }
+};
+
+/*
+Create a tracer with a static life cycle to trace all device kernel execution
+during the program. The unique pointer is used to manage class creation and
+recycling. When the pointer is destroyed, the tracer's destructor will be called
+and print the profile result.
```

Comment:
Could you please explain how it works or add a link to a documentation? Now it seems like a dead code.

Reply:
The static variable lifetime could be referred to https://en.cppreference.com/w/cpp/language/storage_duration. The `tracer`(https://en.cppreference.com/w/cpp/language/raii) will be initialized(call `enableProfiling`) before calling the program's main function and be destroyed(call `disableProfiling`) when the program is about to finish. The detailed implementation of these two functions could refer to https://github.com/intel/pti-gpu/blob/master/samples/cl_hot_kernels/cl_kernel_collector.h which registers some callback function to OpenCL call. We have `clang-tidy` in our CI and will report an error if it is dead code. Do you have any suggestion to make it less confusing(where to insert enabelProfiling and disableProfiling) and I could change it accordingly?

Reply:
What happens when the Enable/DisableProfiling functions are called? There is no implementation in this PR. Please, add comments with the explanation.

Reply:
Thanks for the review! The explanation comment has been added and please help review it again